src/sprightly.ts

```diff
@@ -1,52 +1,43 @@
-import { readFileSync } from "fs"
+import { readFileSync, existsSync } from "fs"
 import path from "path"
 
 interface Data {
   [key: string]: string | number | Array<string | number | Data> | Data
 }
 
-class KeyNotFoundError extends Error {
-  readonly key: string
-  readonly src: string
-  constructor(key: string, src: string) {
-    const message = `Key "${key}" not found in "data" at ${src}`
-    super(message)
-    this.key = key
-    this.src = src
-  }
-}
-
-class PathNotFoundError extends Error {
-  readonly component: string
-  readonly src: string
-  constructor(component: string, src: string) {
-    const message = `Component "${component}" was not found in ${src}`
-    super(message)
-    this.component = component
-    this.src = src
-  }
-}
-
 interface Options {
   keyFallback?: string
   throwOnKeyNotfound?: boolean
   cache?: boolean
 }
 
-function parse(file: string, data: Data, options: Options) {
+function parse(file: string, entryPoint: string, data: Data, options: Options) {
   const parsedFile = file.replaceAll(
     /\{\{(>?)(.*?)\}\}/g,
     (_, isComponent: string, reference: string) => {
       reference = reference.trim()
       if (isComponent) {
-        try {
-          return sprightly(reference, data, options)
-        } catch (e) {
-          throw new PathNotFoundError(reference, "")
+        // if last segement of path is a file
+        // we go back one step to get the directory of the file
+        if (path.extname(entryPoint)) {
+          entryPoint = path.normalize(path.join(entryPoint, ".."))
         }
+
+        const componentAbsolutePath = path.resolve(
```

Comment:
You can remove the above and do this here: `path.join(path.dirname(filename), path.normalize(reference))`.

Reply:
Here you can assume that parse only gets absolute file names. It is like its part of its API. unlike `sprightly`, which might get relative entry points or absolute ones and it has to handle both. Again, it is like its part of its API.